### DISCUSSION TITLE
Getting invalid handshake response getStatus 999 unknown

### DISCUSSION THREAD

**Discussion by CDonalO** [cc-tweaked/CC-Tweaked#651] [Q&A]
`local ws, err = http.websocket("ws://ipaddress:port")`
`if not ws then`
 `print(err)`
`else`
`ws.send("hello")`
`end`
`ws.close()`

im using a linode linux server with a java websocket server on it
 
the java logs that the client is connecting but my turtle gives me an error Getting invalid handshake response getStatus 999 unknown

**Comment by SquidDev** [comment]
999 means that Netty can't parse the HTTP response at all. It's probably worth seeing if you connect to the server using another websocket client (for instance, the browser). If so, can you post the library/code you're using to create a websocket, and I can try to debug.

**Reply by CDonalO** [reply]
i ended up making a javascript ws server and that worked

**Comment by Lemmmy** [comment]
Are you connecting to a HTTPS websocket server instead of a HTTP one? Try using `wss://` instead of `ws://`.

**Reply by CDonalO** [reply]
I tried wss and ws but none worked i ended up making a javascript ws server and that worked
